src/csvcubedpmd/csvwcli/pull.py

```diff
@@ -134,20 +135,24 @@ def _get_rdf_file_dependencies(metadata_file: str) -> List[str]:
 
 
 def _get_table_group_for_metadata_file(metadata_file: str) -> Dict:
-    if isinstance(metadata_file, str):
+    if isinstance(metadata_file, str) and _looks_like_uri(metadata_file):
         return requests.get(metadata_file).json()
```

Comment:
Always check if a request succeeded before using it, i.e:

```python

from requests.exceptions import HTTPError
r = requests.get(metadata_file)
if not r.ok:
    raise HTTPError(f'Failed to get url {metadata_file} with status code {r.status_code}')
```

`r.ok` is just short hand for 2xx status codes, i.e it worked.

I am aware that's pre-existing code but its worth doing, otherwise you end up with rather confusing json parsing errors where a resource can't be reached.

note - this is happening in a few places in this file.